src/views/modules/Player.js

```diff
@@ -173,7 +173,12 @@ Player.prototype.addTracks = function(tracks, noUpdate) {
     return;
   }
 
-  this.tracks = [].concat(this.tracks, tracks);
+  tracks.forEach((track) => {
+    if (!this.tracks.includes(track)) {
```

Comment:
LOL? Really ? Can you use Array.inclues() for this purpose ? I think it's only for String right ? Right now we are holding our own special Object within the array, so you need more checks I think.

Reply:
Fixed!

Reply:
works for identical objects but I see what you mean so fixed!